axelor-base/src/main/java/com/axelor/apps/base/service/PartnerRegistrationCodeServiceImpl.java

```diff
@@ -0,0 +1,122 @@
+package com.axelor.apps.base.service;
+
+import com.axelor.apps.base.ResponseMessageType;
+import com.axelor.apps.base.db.Address;
+import com.axelor.apps.base.db.Country;
+import com.axelor.apps.base.db.Partner;
+import com.axelor.apps.base.db.PartnerAddress;
+import com.axelor.apps.base.db.RegistrationNumberTemplate;
+import com.axelor.apps.base.db.repo.PartnerRepository;
+import com.axelor.apps.base.service.exception.TraceBackService;
+import com.axelor.common.StringUtils;
+import com.google.common.base.Strings;
+import java.util.Optional;
+
+public class PartnerRegistrationCodeServiceImpl implements PartnerRegistrationCodeService {
```

Comment:
it's better to split services by action here, so please rename it to `PartnerRegistrationCodeViewService` and move the method `getRegistrationNumberValidatorClass` to a new service `PartnerRegistrationCodeValidatorFactory`